apps/next/src/pages/Bridge/contexts/BridgeStateProvider/hooks/useTargetToken.ts

```diff
@@ -0,0 +1,47 @@
+import { BridgeAsset, TokenType } from '@avalabs/bridge-unified';
+import { findMatchingBridgeAsset } from '@core/common';
+import { useMemo } from 'react';
+import { useNetworkTokens } from './useNetworkTokens';
+
+export function useTargetToken(
+  targetNetworkId: string,
+  assets: BridgeAsset[],
+  selectedAsset: BridgeAsset | undefined,
+) {
+  const tokens = useNetworkTokens(targetNetworkId);
+
+  const matchedAssets = useMemo(
+    () =>
+      selectedAsset
+        ? assets.filter(
+            (a) =>
+              a.symbol === getWrappedSymbol(selectedAsset) ||
+              a.name === selectedAsset.name,
+          )
+        : [],
+    [assets, selectedAsset],
+  );
+
+  const matchedToken = useMemo(
+    () =>
+      selectedAsset
+        ? tokens.find((t) => findMatchingBridgeAsset(matchedAssets, t))
+        : undefined,
+    [selectedAsset, tokens, matchedAssets],
+  );
+
+  return matchedToken;
+}
+
+const getWrappedSymbol = (asset: BridgeAsset) => {
+  if (asset.type !== TokenType.NATIVE) {
+    return asset.symbol;
```

Comment:
I hate we had to do it this way for Ethereum, but I think we need another `if` for the opposite situation: `WETH.e` -> `ETH` 😱 

Inside the `asset.type !== TokenType.NATIVE` `if` clause:

```ts
    if (asset.symbol === 'WETH.e') {
      return 'ETH';
    }
```

Fortunately this is only a presentational issue (won't actually change the transaction's outcome), so we don't need to worry about safety here (other than someone creating an `ETH` token with some obscene logo).